### DISCUSSION TITLE
Email Data Type

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#543] [Maintainer discussions]
The specs related to issue #235  are ready for review. Please take a look at the attached document and follow the [review guidelines](https://wiki.mathesar.org/en/design/process/review-guidelines).

## Document Link
https://wiki.mathesar.org/en/design/specs/email-data-type

@mathemancer @kgodey @pavish, your feedback is required by Wednesday 18th. Please plan your review so that at least one round of changes can be made and reviewed if needed.

**Comment by kgodey** [comment]
@ghislaineguerin Looks good! Feedback:

(1) It would be good to have a list of all email filtering options somewhere, either in the prototype or the spec text document. The frontend needs to have a full list of what to implement.

(2) Do we want emails to be clickable (with `mailto` links?) If so, we should format them as links.

(3) I think the empty row for "add row" should be more visually distinct somehow. It looks confusing when you're grouping since it looks just like any other row.

**Reply by ghislaineguerin** [reply]
@kgodey I've added the filter list and styled emails as links. I agree with the empty row not being distinct enough. Is this something we can solve during UI revamp?

**Reply by kgodey** [reply]
The email changes look good.

I think it's fine to wait until we revamp the UI to make changes to the empty row's design.

**Comment by kgodey** [comment]
@ghislaineguerin Separate question: all your review requests contain this line: 

> Please plan your review so that at least one round of changes can be made and reviewed if needed.

I'm not sure what this means, does this mean that we need to review the document before the deadline listed in the previous sentence? What's the action item for reviewers here?

**Reply by ghislaineguerin** [reply]
It's mostly that the review should be planned, as in, take a quick look at the overall design in case there is some big inconsistency or something that will require more than just minor tweaks, and plan to communicate that in advance if possible. Could be worded better.

**Reply by kgodey** [reply]
Perhaps you could word it something like:

> ### Reviewers
> [USERNAMES], your feedback is required, everyone else is also welcome to provide feedback.
>
> ### Deadlines
> Please take a quick look at the spec by [EARLY DEADLINE] and let me know if there are any major structural flaws with the design. Comprehensive feedback is due by [LATE DEADLINE].

I added a line about everyone else being welcome to provide feedback to invite feedback from the community as well.

**Comment by ghislaineguerin** [comment]
@pavish @kgodey 
The specs have been updated to include the global data type options. Minimal changes were introduced to email types, as there were no options other than the default value.

**Reply by kgodey** [reply]
Looks good, thanks!
